File: packages/test-kit/fixtures/fastapi-app/app/routers/deals.py
from fastapi import APIRouter

# Renamed on purpose (epic-16 line 140): the module defines `Deal`, this file calls it
# `DealSchema`. Renaming a binding must not change the graph — the CALLS edges below still have to
# land on `app/models.py#Deal`. Before the assembler translated a local alias back to the exported
# name it looked `DealSchema` up in that module's export table, found nothing, and dropped them.
from app.models import Deal as DealSchema

router = APIRouter(tags=["deals"])


@router.get("/")
def list_deals() -> list[DealSchema]:
    return [DealSchema(id="d1", name="Project Alpha")]


@router.get("/{deal_id}")
def get_deal(deal_id: str) -> DealSchema:
    return DealSchema(id=deal_id, name="Project Alpha")


@router.post("/")
def create_deal(deal: DealSchema) -> DealSchema:
    return deal

File: packages/test-kit/fixtures/ts-basic/src/api/deals.ts
// The import is RENAMED on purpose (epic-16 line 140). Renaming a binding must not change the
// graph: the CALLS edge below still has to land on `createRepository` in ../lib/deal-repository,
// because that is the symbol this file calls. Before the assembler learned to translate a local
// alias back to the exported name it looked `buildRepository` up in that file's export table,
// found nothing, and dropped the edge — so this golden losing a line is the regression signal.
import { createRepository as buildRepository } from '../lib/deal-repository';

export function getDeals(): string[] {
  return buildRepository().findAll();
}
